### ISSUE TITLE
Improve usability of LHS navbar

### ISSUE THREAD

**Issue by inglesp** [opensafely/documentation#1270]
1) Sometimes the LHS navbar automatically scrolls to the right position (eg [here](https://docs.opensafely.org/ehrql/tutorial/running-ehrql/)) and sometimes it doesn't (eg [here](https://docs.opensafely.org/ehrql/)).  I think this is related to whether the page is a leaf node in the hierarchy or not.

2) The current page in is indicated in the LHS navbar by an easy-to-miss underlining.  Could we change this to something more obvious?

**Comment by iaindillingham** [comment]
Following #1258, the LHS navbar is expanded and contains the entire navtree (`mkdocs.yml.nav`). This makes it hard to navigate the documentation; as a demonstration, measure the time it takes to find the ehrQL overview with `navigation.expand` set and unset. (Spoiler: even when I know what link to click, the former takes 10s; the latter takes 1s.)

Expanding the LHS navbar was a solution to the problem of "show the Internet how much documentation we have". However, it has come at the cost of making it hard to navigate that documentation. We know this not just through the above anicdata, but also from user-feedback (Rose).


Could we revisit the solution?

Alternative candidate solutions:

* display the navtree on a sitemap page (remember those?)
* display the navtree on the homepage
* add a, "Expand" link to the navtree

**Comment by inglesp** [comment]
We've decided to revert the change.

**Comment by sebbacon** [comment]
This came as a direct in-person request from @Bengoldacre so it's important to explain this decision to him.

**Comment by iaindillingham** [comment]
> This came as a direct in-person request from @Bengoldacre so it's important to explain this decision to him.

Are you able to do this, @inglesp?

**Comment by inglesp** [comment]
In [this thread](https://bennettoxford.slack.com/archives/C31D62X5X/p1689688107233569) which I missed first time around, Steve made some other suggestions and Ben picked one that works for him.  (Enable `navigation.sections` and disable `navigation.expand`.)

@iaindillingham could you try this?

**Comment by iaindillingham** [comment]
Let's compare and contrast. First, this is where we are now; that is, `navigation.expand` is enabled.

![expand](https://github.com/opensafely/documentation/assets/477263/ebb5af0f-5bc0-4913-a10d-0b4a78b5a1c4)

Next, let's disable `navigation.expand` and enable `navigation.sections`. Notice that there are the same entries in the TOC; they're just not indented.

![sections](https://github.com/opensafely/documentation/assets/477263/4abf34e3-bb22-4aa4-bc7c-b1bc21b1e696)

Finally, let's return to the default; that is, both `navigation.expand` and `navigation.sections` are disabled.

![default](https://github.com/opensafely/documentation/assets/477263/f6e44940-5051-4042-abd8-b003c4ddec4f)

If the LHS navbar contains the entire nav tree, then it is hard to navigate the documentation.

**Comment by StevenMaude** [comment]
> display the navtree on a sitemap page (remember those?)

For what it's worth, there is already an [XML sitemap](https://docs.opensafely.org/sitemap.xml) if the goal is just to see how many pages there are, that could suffice, although it's unformatted XML.

> If the LHS navbar contains the entire nav tree, then it is hard to navigate the documentation.

I agree with this. I made the initial change because it was requested.

***

Confusingly, it is possible to set a depth for the table of contents on the right-hand side, that is, the page navigation. But not for the left-hand site navigation. (This is because python-markdown generates the right-hand page table of contents, but not the left-hand.)

I also tried quickly trying to add an additional template to `overrides/` to generate a full navigation list, but couldn't easily get that to work. I might be missing something in how the templates work though.
